process/interceptors/processor/miniblockInterceptorProcessor.go

```diff
@@ -71,11 +77,26 @@ func (mip *MiniblockInterceptorProcessor) Save(data process.InterceptedData, _ p
 		return err
 	}
 
+	isCrossShardForMe := mip.isMbCrossShardForMe(miniblock)
+	isWhiteListed := mip.whiteListHandler.IsWhiteListed(data)
+	if isCrossShardForMe && !isWhiteListed {
+		log.Debug("miniblock interceptor processor : cross shard miniblock for me",
+			"message", "not whitelisted will not be added in pool")
+		return nil
+	}
+
 	mip.miniblockCache.HasOrAdd(hash, miniblock)
 
 	return nil
 }
 
+func (mip *MiniblockInterceptorProcessor) isMbCrossShardForMe(miniblock *block.MiniBlock) bool {
+	isCrossShard := miniblock.SenderShardID != mip.shardCoordinator.SelfId()
```

Comment:
We have to be more specific here: we might end up having miniblocks with senderID = [some hardcoded value such as metachainID or AllShardID] and receiverID = [AllShardID]

Reply:
receiver is checked in the following line to be selfId

Reply:
How I see it is as follows: This function should return true whenever the miniblock sender shard id is not from the current shard. If the sender is from the current shad (either has the same shard ID or is AllShardID) than this function should return false in order to invalidate the if instruction on line 82 and add the miniblock in pool. If the sender is not the current shard should return false (regardless the receiver shard ID) because it is a miniblock not originated from this shard and should be checked against the whitelist mechanism in order to be added to the pool. This way ensures that he cross shard miniblocks with destination me are handled the same as any miniblock requested by the hardfork mechanism and that is: check if it is in whitelist.
So I would change into, as of today:
```GO
func (mip *MiniblockInterceptorProcessor) isMbCrossShardForMe(miniblock *block.MiniBlock) bool {
        return miniblock.SenderShardID == mip.shardCoordinator.SelfId() 
}
```

Reply:
done.